opendevin/controller/agent_controller.py

```diff
@@ -101,7 +101,10 @@ async def _run(self):
                     self._task_state = TaskState.FINISHED
             except Exception as e:
                 logger.error('Error in loop', exc_info=True)
-                raise e
+                await self._run_callbacks(
+                    AgentErrorObservation(f'I got an error, detail info: {str(e)}. The task has stopped.'))
```

Comment:
You are absolutely right that the information that the task has stopped should be displayed to the user. I'm not equally sure about presenting exception detail in the agent's _task_ tab. We had that happen until recently and we changed it... it was surprising to users. The exceptions from running the app are not the same kind of thing with the task-related messages, errors or otherwise, that are expected in that box.

I wonder if we could have a 'system status' field somewhere, for example near the settings 'wheel' button, where the user gets this kind of info, at least the exception name or, as possible, a quick clue "check your authentication settings"...?

Reply:
I think we should keep the message vague, and send the user to the backend logs. Sending raw errors to the frontend can cause issues e.g. with token leakage (like in the screenshot in this PR 😄)